### PR TITLE
Docker Image for Website Dev

### PR DESCRIPTION
**Pull request by pmbaumgartner**
Adds a `Dockerfile` for a repeatable build of the website with a working version of node/npm and instructions for accessing the images through the github container registry for others to use.

### Types of change
Documentation

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: website/README.md [hunk 1]
```diff
@@ -554,6 +554,40 @@ extensions for your code editor. The
 [`.prettierrc`](https://github.com/explosion/spaCy/tree/master/website/.prettierrc)
 file in the root defines the settings used in this codebase.
 
+## Building & Developing the Site with Docker {#docker}
```
Comment (svlandeg) [inline website/README.md:557]: Nitpick: I wouldn't use capitals across the title, most other sections in this readme only capitalize the first letter
```suggestion
## Building & developing the site with Docker {#docker}
```

File: website/README.md [hunk 2]
```diff
@@ -554,6 +554,40 @@ extensions for your code editor. The
...
+  -p 8000:8000 \
+  ghcr.io/explosion/spacy-io \
+  gatsby develop -H 0.0.0.0
+```
+
+This will allow you to access the built website at http://0.0.0.0:8000/ 
+in your browser, and still edit code in your editor while having the site 
+reflect those changes.
+
+**Note**: On M1 Macs you may need to the image tagged `arm64` (`ghcr.io/explosion/spacy-io:arm64`),
```
Comment (svlandeg) [inline website/README.md:578]: there's a verb or something missing here?

File: website/README.md [hunk 3]
```diff
@@ -554,6 +554,40 @@ extensions for your code editor. The
...
+otherwise you'll see `qemu` segfault during the build.
+
+### Building the Docker Image {#docker-build}
```
Comment (svlandeg) [inline website/README.md:581]:
```suggestion
### Building the Docker image {#docker-build}
```

**Review by svlandeg — COMMENTED** [review]
This will be great to have, thanks so much for looking into this Peter!
I only had some small nitpicking comments :-)

File: website/README.md (hunk 2 above)
Comment (pmbaumgartner) [reply website/README.md:578]: It makes sense to me but it came from my brain so that's not a great test :) I always favor more clarity. How about:

> If you're working on a Mac with an M1 processor, you might see segfault errors from `qemu` if you use the default image. To fix this, use the `arm64` tagged image in the `docker run` command (`ghcr.io/explosion/spacy-io:arm64`).

Comment (svlandeg) [reply website/README.md:578]: so it was missing the verb "use"? (I'm trying really hard to parse your original sentence but I really can't, haha, it may just be me)

Comment (pmbaumgartner) [reply website/README.md:578]: I'll reformulate to my new suggestion!

To me it's implied that they would "use" it in the original sentence, like if I gave someone an umbrella before they went outside I'd say "You might need this".

Comment (svlandeg) [reply website/README.md]: But but but it said "may need to the image", not "may need the image" 🙈 
(it's not relevant, at all, I'm just so confused)

Comment (pmbaumgartner) [reply website/README.md]: Well then this is definitely a case of me failing to read my own writing correctly 🙈

**Review by ines — APPROVED** [review]
